### ISSUE TITLE
[ONGOING] District & State Attorney Data

### ISSUE THREAD

**Issue by billimarie** [billimarie/prosecutor-database#80]
Please see the [README.md](https://github.com/billimarie/prosecutor-database/blob/master/README.md) before replying.

Once you've finished populating your sheet, comment below.

**Comment by syeedameen** [comment]
Greetings of the day madam , I am a beginner in open source .
I would like to work on this.
And I am also taking part in Hacktoberfest , So can I contribute to it in October.

**Comment by billimarie** [comment]
Hi @syeedameen, thank you for reaching out. You are welcome to join us as we embark upon Prep-tember & Hacktoberfest. Feel free to follow the instructions listed above, then submit your pull request. If you have any questions, I can be reached here or through email (link in profile).

**Comment by andrefpoliveira** [comment]
Hey! I would like to work on this for the Hacktoberfest if possible 😁

**Comment by billimarie** [comment]
Hi @andrefpoliveira, thanks for reaching out. It's a collaborative issue so please feel free to tackle it. Let me know if you have any questions about setting up.

**Comment by dylancaponi** [comment]
Step 1 worked with `git clone https://github.com/billimarie/prosecutor-database.git`

**Comment by dylancaponi** [comment]
@billimarie What's the best way to just update missing photos or profiles with partial information?

**Comment by jSadoski** [comment]
@dylancaponi If you feel comfortable diving into the .json file where the information exists and changing the values there, `git` will track those changes so you can upload them to GitHub!

**Comment by dylancaponi** [comment]
@jSadoski I'd much rather do that but I don't see the json files in master or hacktoberfest branches.

I do see them in PRs that were merged like this one: https://github.com/mattkduran/prosecutor-database/tree/master/csv-json-data

Where is that data getting merged?

**Comment by jSadoski** [comment]
I was just noticing in the project that I think the data was restructured. It's now just .js files under `./api/` I believe.

**Comment by billimarie** [comment]
@dylancaponi @jSadoski Yes, the data is vetted first through this repo under the `api` folder. Once it has been verified, I import the new entries into a MongoDB database, which the live Heroku instance then pulls from.

**Comment by shariq1989** [comment]
FYI for newcomers. If you fail on _step 3_ after running _git fetch_ like below:
`git fetch origin hacktober2020
fatal: Not a git repository (or any of the parent directories): .git`

You have to run `cd prosecutor-database\` and then run the _git fetch_ command

**Comment by fedGL** [comment]
I've collected information and I'm ready to commit my CSV!

**Comment by fedGL** [comment]
Question: I cloned the repo and I got my CSV ready to convert to JSON as it says in the instructions. Where should I put this? I see a folder called "🔥 100 DISTRICT ATTORNEYS 🔥" with a folder inside named "Your-State-Here" and csv and json files inside. I'm assuming that's the correct place?
Also, for the "Your-State-Here", what convention are you following for states with two words? Do we hypen them with - ?

**Comment by billimarie** [comment]
@shariq1989 Great tip! If you have the bandwidth, can you update the [HACKTOBER2020.md](https://github.com/billimarie/prosecutor-database/blob/master/HACKTOBER2020.md) file with those instructions?

@fedGL Wonderful. You can place it in the `100 District Attorneys` folder & stick with the abbreviated states (ex: "CA" or "NY"). I will update the instructions to show the naming conventions.

**Comment by janel-developer** [comment]
Hi @billimarie - I was just having a poke around the internet to see if I could find any data sources we could use and I found this api - https://www.courtlistener.com/api/rest-info/

Have you seen it? I thought I might send them a message and ask about access to the attorneys endpoint - which doesn't have public access. I thought I'd ask them what kind of data is there. I looked at the people and positions endpoints, but they don't seem to list district attorneys - at least not as district attorneys - looking at the possible values for position_type (attached here from a query)

[position_types.txt](https://github.com/billimarie/prosecutor-database/files/5325436/position_types.txt)

**Comment by billimarie** [comment]
Hello @janel-developer, thank you for researching data sources--would make this project a lot easier. I had not come across that api. It would be amazing if you could reach out. Would you like me to assign a dedicated issue for you? I appreciate your research on this.

**Comment by janel-developer** [comment]
Happy to help! I've sent a message to see what I can find out. 

It would be great to use a dedicated issue for this. I'I'd be happy to write something that just pulls and formats data as json on the server side.

**Comment by billimarie** [comment]
Wonderful--when you have a moment, can you leave a comment on the issue so I can assign it to you? See: #145

**Comment by lisaood** [comment]
Hello, I am a student in 3rd year of computer science license degree, I would like to contribute to your project . thank you

**Comment by k-garrido** [comment]
Hi there! 
My is Katherinne Garrido, I'm student of Laboratoria and I really would like to contribute to this project, will there be another hacktober for 2021?
Greetings :)

**Comment by HallowedDust5** [comment]
Was this only for Hacktober 2020, or is this issue still open?

**Comment by mahiaash** [comment]
Hi, 
I'm Mahia Asha and I was hoping to add to the project? Can I add more district attorneys or complete ones that exist? Is there anyway I could help? Please let me know!

**Comment by billimarie** [comment]
Bringing this issue back to life; new goal is to integrate the previous `hacktober-data` into our current `main` branch.
